build.gradle

```diff
@@ -106,5 +132,310 @@ subprojects { project ->
+        isExample = subproject.name.startsWith('example')
+        isGrailsPlugin = subproject.name.startsWith('grails-plugin')
+        pomInfo = {
+            delegate.name subproject.title
+            delegate.description subproject.projectDesc
+            delegate.url projectUrl
+
+            delegate.licenses {
+                delegate.license {
+                    delegate.name 'The Apache Software License, Version 2.0'
+                    delegate.url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    delegate.distribution 'repo'
+                }
+            }
+
+            delegate.scm {
+                delegate.url "scm:git@github.com:${githubSlug}.git"
+                delegate.connection "scm:git@github.com:${githubSlug}.git"
+                delegate.developerConnection "scm:git@github.com:${githubSlug}.git"
+            }
+
+            if (developers) {
+                delegate.developers {
+                    for (dev in developers.split(',')) {
+                        delegate.developer {
+                            delegate.id dev.toLowerCase().replace(' ', '')
+                            delegate.name dev
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (!isExample) {
+
+        apply plugin: 'maven-publish'
+        apply plugin: 'signing'
+
+        Task sourcesJar = subproject.tasks.findByName("sourcesJar")
+        if (!sourcesJar) {
+            sourcesJar = tasks.create("sourcesJar", Jar) {
+                archiveClassifier = 'sources'
+                group = BasePlugin.BUILD_GROUP
+                description = 'Assembles a jar archive containing the main sources of this project.'
+                from subproject.sourceSets.main.allSource
+            }
+        }
+
+        Task javadocJar = subproject.tasks.findByName("javadocJar")
+        if (!javadocJar) {
+            javadocJar = tasks.create("javadocJar", Jar) {
+                archiveClassifier = 'javadoc'
+                group = BasePlugin.BUILD_GROUP
+                description = 'Assembles a jar archive containing the generated Javadoc API documentation of this project.'
+                from subproject.plugins.hasPlugin(GroovyPlugin) ? subproject.tasks.getByName(GroovyPlugin.GROOVYDOC_TASK_NAME) : subproject.tasks.getByName(JavaPlugin.JAVADOC_TASK_NAME)
+            }
+        }
+        tasks.findByName("assemble").finalizedBy(sourcesJar, javadocJar)
+
+        publishing {
+            if (isSnapshot) {
+                repositories {
+                    maven {
+                        credentials {
+                            def u = System.getenv("ARTIFACTORY_USERNAME") ?: subproject.hasProperty("artifactoryPublishUsername") ? subproject.artifactoryPublishUsername : ''
+                            def p = System.getenv("ARTIFACTORY_PASSWORD") ?: subproject.hasProperty("artifactoryPublishPassword") ? subproject.artifactoryPublishPassword : ''
```

Comment:
Change `subproject.hasProperty('x') ? subproject.x : ''` to `project.findProperty('x') ?: ''`

Reply:
Updated